nifi-api/src/main/java/org/apache/nifi/processor/AbstractProcessor.java

```diff
@@ -27,7 +30,12 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
             onTrigger(context, session);
             session.commit();
         } catch (final Throwable t) {
-            getLogger().error("{} failed to process due to {}; rolling back session", new Object[]{this, t});
+            StringWriter stacktraceWriter = new StringWriter();
```

Comment:
I wonder if it would make sense to surface the shorthand message to the user in the UI but log the stacktrace to the log file. I imagine this would take some re-working.